source/linux/Foundational_Components/System_Security/SELinux.rst

```diff
@@ -0,0 +1,139 @@
+
+Filesystem Setup
+----------------
+
+Yocto
+^^^^^
+
+The following steps describe how to build SELinux user-space tools and configuration on Yocto. Please use :ref:`Processor SDK - Building the SDK with Yocto <building-the-sdk-with-yocto>` as reference.
+
+1. On a fresh build, update the config file to use :file:`configs/arago-scarthgap-selinux-config.txt`. This config includes the opensource `meta-selinux` layer into the build and also adds additional features for the filesystem.
+
+   .. note::
+
+      The default policy is targeted. If you want to update it, add the variable ``PREFERRED_PROVIDER_virtual/refpolicy = "refpolicy-<your_choice>"`` to the :file:`local.conf` where ``<your_choice>`` is targeted, mls, mcs.
+
+2. Build your custom image in build directory
+
+   .. code-block:: console
+
+      $ MACHINE=<machine> bitbake -k tisdk-default-image
+
+3. Flash the generated wic image onto your SD card. By default, SELinux functionality is in the disabled state
+
+Debian
+^^^^^^
+
+The following steps describe how to install and configure SELinux together with the default policy which enables SELinux for the most important parts of your Debian filesystem. This guide does not describe how to build the filesystem using Debian.
+
+1. The default Debian image provided by TI (:file:`tisdk-debian-trixie-<machine>-<version>.wic.xz`) packages necessary kernel configs to enable SELinux. If you are not using the standard linux kernel provided by debian with an ext2/3/4/xfs/btrfs filesystem, you have to make sure that you are using a SELinux capable kernel and filesystem. For ext2/3/4 there is nothing special to be aware of, but for other filesystems there are still some quirks.
+
+2. On the target machine, get the default policy and the basic set of SELinux utilities by running
+
+   .. code-block:: console
+
+      $ sudo apt-get install selinux-basics selinux-policy-default auditd
+
+Permissive mode Setup
+---------------------
+
+It is intended to go into Permissive mode, log the accesses, create the access policy and only then jump to Enforcing mode. Skipping this step would lead to a login failure due to lack of access.
+
+1. Boot the target (in disabled state), Run ``fixfiles -F onboot`` or ``touch /.autorelabel`` to ensure that files are relabeled upon next reboot.
+
+   .. warning::
+
+      When systems run SELinux in permissive mode, users and processes might label various file-system objects incorrectly. File-system objects created while SELinux is disabled are not labeled at all. This behavior causes problems when changing to enforcing mode because SELinux relies on correct labels of file-system objects.
+
+   To prevent incorrectly labeled and unlabeled files from causing problems, SELinux automatically relabels file systems when changing from the disabled state to permissive or enforcing mode.
+
+2. Open :file:`/etc/selinux/config` and edit it to ``SELINUX=permissive`` and the policy type you are using.
+
```

Comment:
I would not recommend adding any further information on this concept since we do not want to support queries about policy types. This guide seeks to explain how to enable SELinux, not serve as an introduction to the concept.

However, in future if we do decide to add it, I would also recommend adding in how to write a custom policy and fix policy errors.

Reply:
I will move this to resolved for now, but you can bring it up again